spec/integration/admin_censor_rule_spec.rb

```diff
@@ -0,0 +1,37 @@
+# -*- encoding : utf-8 -*-
+require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
+require File.expand_path(File.dirname(__FILE__) + '/alaveteli_dsl')
+
+describe 'Adding an authority censor rule' do
+  before do
+    allow(AlaveteliConfiguration).to receive(:skip_admin_auth).and_return(false)
+
+    confirm(:admin_user)
+    @admin = login(:admin_user)
+    @authority = FactoryGirl.create(:public_body)
+  end
+
+  it 'clears the cache for existing info_requests' do
+    request = FactoryGirl.create(:info_request_with_plain_incoming,
+                                 :public_body => @authority)
+    incoming_message = request.incoming_messages.last
+    incoming_message._cache_main_body_text
+
+    @authority.reload
+    incoming_message.reload
+
+    expect(incoming_message.cached_main_body_text_unfolded).to_not be_nil
+
+    using_session(@admin) do
+      visit new_admin_body_censor_rule_path(@authority)
+      fill_in 'censor_rule_text', :with => 'rubbish question'
+      fill_in 'censor_rule_replacement', :with => '[REDACTED]'
+      fill_in 'censor_rule_last_edit_comment', :with => 'test'
+      click_button 'Create'
+    end
+
+    incoming_message.reload
+    expect(incoming_message.cached_main_body_text_unfolded).to be_nil
```

Comment:
I think I'd feel better with an even more explicit spec here.

* create request with specific text e.g. "I have a rubbish question for you"
* add new censor rule as you've done
* visit request page
* check that request page has the contents "I have a [REDACTED] for you"